sycl/test/check_device_code/group_operations.cpp

```diff
@@ -1,18 +1,19 @@
-// RUN: %clangxx -I %sycl_include -S -emit-llvm -fsycl-device-only %s -o - | FileCheck %s
+// RUN: %clangxx -I %sycl_include -S -emit-llvm -fsycl-device-only -ffp-model=precise %s -o - | FileCheck %s
 
 #include <CL/sycl.hpp>
 
 using namespace sycl;
 
 template <typename T, template <typename> class O, typename G>
-void group_operations(G g) {
+[[gnu::always_inline]] void group_operations(G g) {
```

Comment:
I believe this doesn't force the inlining.
A predictable way would be to "disable" inlining, I think

Reply:
> A predictable way would be to "disable" inlining, I think

Since I am checking for the `__spirv_GroupXxx` calls, it is much easier when they are all in the expected order with the argument values directly there. Without inlining, all the helper functions are preserved, and parameters are used instead of values, making it hard to see repeated calls to the same `__spirv_GroupXxx` function.

> I believe this doesn't force the inlining.

Given the above, can you think of a better way to look at the  `__spirv_GroupXxx` in order and with the values?

Reply:
I don't think you would be able to see values without inlining. I am OK if adding this attribute actually works.
BTW, does it force inlining any better than standard "inline" keyword?

Reply:
The attribute does seem to force inlining better than the standard `inline` keyword, but I am not sure on the details and the reach. What I am seeing with our compiler in this test is that it disables some inlining heuristics, which results in inlining all of the helpers, regardless of the compiler version/build.

Reply:
If just "inline" works too, I'd suggest going with it.
There is no attribute that can 100% force inlining.

Reply:
Makes sense, but that's the thing: `inline` depends on additional heuristics, and our compiler does not inline some of the cases here. The attribute makes it go "all the way" here, which I guess is due to disabling some of those heuristics.